pybamm/meshes/one_dimensional_submeshes.py

```diff
@@ -34,7 +34,7 @@ def __init__(self, edges, coord_sys, tabs=None):
         self.internal_boundaries = []
 
         # Add tab locations in terms of "left" and "right"
-        if tabs:
+        if tabs and "negative tab" not in tabs.keys():
             self.tabs = {}
```

Comment:
I'm not sure I understand what is going on here. Why are we checking that `negative tab` isn't in the dict?

Reply:
At the point of mesh initialisation, if tabs have been provided, they look something like
`{"negative": {"z_centre": 0}, "positive": {"z_centre": 1}}`.
During the init function they are converted so that self.tabs = e.g. 
`{"negative tab": "left, "positive tab": "right"}`,
so an initialised mesh['tabs'] will have `negative tab` in the dict.

The former format is never saved, and the serialiser only ever a fully initialised mesh, i.e. the point where the tab locations have been fully determined, so that it what's written out to the json file. When reading back in and hitting the __init__ function to re-create the mesh, we search for whether the lab location has already been determined (by the presence of `negative tab` vs `negative` in the tab dict) and proceed accordingly.